### PR TITLE
issue fixed

### PR DESCRIPTION
**Pull request by 1killerbehindyou1**
issue fixed

### CODE REVIEW

File: BasicGUI/QMLSources/main.qml [hunk 1]
```diff
@@ -43,26 +44,31 @@ ApplicationWindow
...
+                        }
+                    }
+                    else 
+                    {
+                        previev_is_active = false;
+                        if(image_selected)
+                        {
+                            loaded_image.visible = false;
+                        }
+                    }
```
Comment (Nutek) [inline BasicGUI/QMLSources/main.qml:68]: There is shorter way to express above code

**Review by Nutek — CHANGES_REQUESTED** [review]
In my opinion there is other way to solve this issue

File: BasicGUI/QMLSources/main.qml [hunk 2]
```diff
@@ -43,26 +44,31 @@ ApplicationWindow
         Menu 
         {
             title: qsTr("&View")
-            Action 
+            delegate: CheckBox
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/main.qml]: Changed

File: BasicGUI/QMLSources/main.qml (hunk 1 above)
Comment (Nutek) [reply BasicGUI/QMLSources/main.qml:72]: Proposal sended via Teams

File: BasicGUI/QMLSources/main.qml [hunk 3]
```diff
@@ -43,24 +45,59 @@ ApplicationWindow
...
             }
-             MenuSeparator { }
-            Action 
+
+            Action
             {
-                text: qsTr("&HideImage before trasformation...")
-                onTriggered:
+                text: "Preview rendered image..."
+                checked: rendered_previev_is_active
```
Comment (Nutek) [inline BasicGUI/QMLSources/main.qml:83]: You need binding `rendered_preview_is_active: actionPreviewShow.checked`

**Review by Nutek — CHANGES_REQUESTED** [review]

File: BasicGUI/QMLSources/main.qml (hunk 1 above)
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/main.qml:62]: Corrected.

File: BasicGUI/QMLSources/main.qml (hunk 3 above)
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/main.qml:73]: Corrected.

Comment (Nutek) [reply BasicGUI/QMLSources/main.qml:73]: Truly you changed code. Unfortunately you still manually flip boolean state.

**Review by Nutek — CHANGES_REQUESTED** [review]

File: BasicGUI/QMLSources/main.qml (hunk 3 above)
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/main.qml]: Corrected.

File: BasicGUI/QMLSources/main.qml [hunk 4]
```diff
@@ -14,8 +14,19 @@ ApplicationWindow
...
-    property bool image_selected: false
+    property bool imageSelected: false
+    property bool previevIsActive: showSelectedImage.checked
+    property bool renderedPrevievIsActive: showRenderedPreview.checked
 
+    onPrevievIsActiveChanged: 
+    {
+        if(imageSelected)
+        {
+            loadedImage.visible = previevIsActive
```
Comment (Nutek) [inline BasicGUI/QMLSources/main.qml:25]: I would rather suggest complex assignement instead `if` statement:
```js
loadedImage.visible = imageSelected && previewIsActive
```

File: BasicGUI/QMLSources/main.qml [hunk 5]
```diff
@@ -14,8 +14,19 @@ ApplicationWindow
     title: qsTr("Basic Gui")
     color: "lightgrey"
 
-    property bool image_selected: false
+    property bool imageSelected: false
+    property bool previevIsActive: showSelectedImage.checked
```
Comment (Nutek) [inline BasicGUI/QMLSources/main.qml:18]: Typo: previe`w`IsActive

File: BasicGUI/QMLSources/main.qml (hunk 4 above)
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/main.qml:25]: Implemented.

**Comment by 1killerbehindyou1** [comment]
Fixed problems with check box in menu and added render preview.

**Pull request merged**